definitions/npm/angular_v1.4.7/flow_v0.47.x-/angular_v1.4.7.js

```diff
@@ -22,12 +29,12 @@ declare module angular {
     injector: Function,
   }
 
-  declare function AngularLinkFunction(
-    scope: mixed,
+  declare type AngularLinkFunction = (
+    scope: $Scope<*>,
     element: JqliteElement,
     attrs: mixed,
     controller: mixed
-  ): void
+  ) => void
```

Comment:
any idea why flow couldn't find `$Scope<*>` when AngularLinkFunction was declared as `declare function`? And whats the actual difference with `declare function fn():void` and `declare type fn = () => void`?

Reply:
I've found it's easier to declare overrides with `declare type` and just intersection them with `&`. Could that be related to what you're looking at?

Reply:
Seems that with `declare function` ordering matters, but not with `declare type fn = () => {}`. Do you know if there are any other differences? One what i'm really curious about is that is it somehow possible to type function `this`. Naturally that's not possible with arrow function, but with regular function libdefs that would make much sense.